lib/ethers/multicall.ex

```diff
@@ -0,0 +1,343 @@
+defmodule Ethers.Multicall do
+  @moduledoc """
+  High-level module providing convenient utilities and an easy-to-use API for interecting with
+  `Multicall3` (more info https://www.multicall3.com).
+
+  This module aggregates multiple **read-only** Ethereum contract calls into a single eth_call
+  which can be passed to `Ethers.call/2` or `Ethers.Multicall.call/2`. The latter is
+  recommended for decoding responses returned by the Multicall contract.
+  """
+
+  import Ethers.Utils, only: [hex_decode!: 1]
+
+  alias Ethers.Contracts.Multicall3
+  alias Ethers.ExecutionError
+  alias Ethers.TxData
+
+  @typep aggregate3_options :: [to: Ethers.Types.t_address(), allow_failure: boolean()]
+  @typep aggregate2_options :: [to: Ethers.Types.t_address()]
+
+  @doc """
+  Aggregate calls, ensuring each returns success if required. Returns an `Ethers.TxData`
+  which can be passed to `Ethers.call/2` or `Ethers.Multicall.call/2`.
+  More info at: https://github.com/mds1/multicall#batch-contract-reads
+
+  ## Parameters
+  - data: A list of function calls with optional options
+    - allow_failure: If false the execution will revert. Defaults to true.
+    - to: Overrides the `default_address` (if any) for the respective function call.
```

Comment:
```suggestion
  - data: A list of function calls with optional options
  - allow_failure: If false the execution will revert. Defaults to true.
  - to: Overrides the `default_address` (if any) for the respective function call.
```

Also in other functions below.

Reply:
`allow_failure` and `to` are both optional, so the indentation is fine here, wdyt?

Reply:
Ah okay. I get it now. Let's add a bit more details

```suggestion
  - data: A list of `TxData` structs or `{%TxData{...}, options}` tuples. The options can include:
    - allow_failure: If false the execution will revert. Defaults to true.
    - to: Overrides the `default_address` (if any) for the respective function call.
```